### ISSUE TITLE
Revisit chain.archiveStateEpochFrequency hidden status

### ISSUE THREAD

**Issue by wemeetagain** [ChainSafe/lodestar#6908]
See https://discord.com/channels/593655374469660673/743858262864167062/1255140271709094050

> `--chain.archiveStateEpochFrequency` still exists, its just a hidden option

> Why hidden? I know of another dev that tried upgrading and that kinda tripped them up, would be useful to know about such an option

It would be good to articulate why this flag is hidden or unhide it.

**Comment by nflaig** [comment]
shouldn't be hidden, it also gives more visibility on how many states we store by default which is otherwise undocumented

**Comment by claravanstaden** [comment]
@silentfred you see. 😉

**Comment by philknows** [comment]
This is the [last comment](https://github.com/ChainSafe/lodestar/pull/5154#discussion_r1111139220) made about why it's hidden. Has your stance changed on this @wemeetagain ?

**Comment by wemeetagain** [comment]
No. This issue will go away when we get https://github.com/ChainSafe/lodestar/pull/7005
